File: attendancer/backend/AttenDancer.Entity/Dtos/Event/EventViewDto.cs
using AttenDancer.Entity.Dtos.Participant;
using AttenDancer.Entity.Entity_Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AttenDancer.Entity.Dtos.Event
{
    public class EventViewDto
    {
        [Required]
        public string Name { get; set; } = null!;

        [Required] 
        public string Id { get; set; } = null!;

        [Required]
        public string Date { get; set; } = null!;

        [Required]
        public string UserId { get; set; } = null!;

        [Required]
        public string? QrCode { get; set; } = null!;

        [Required]
        public bool IsQrValid { get; set; } = true;

        [Required]
        public List<string>? Metadata { get; set; }

        public string? EventGroupName { get; set; }

        public string? EventGroupId { get; set; }

        [Required]
        public ICollection<ParticipantViewDto>? Participants { get; set; } = new List<ParticipantViewDto>();
    }
}

File: attendancer/frontend/src/app/models/event-view-dto.ts
import { ParticipantsViewDto } from "./participants-view-dto"

export class EventViewDto {
    id: string = ""
    name: string = ""
    date: string = ""
    userId: string = ""
    qrCode: any = null
    isQrValid: boolean = false
    metadata: string[] = []
    eventGroupName: string = ""
    eventGroupId: string = ""
    participants: ParticipantsViewDto[] = []
}
